core-libs/setup/ssr/logger/services/express-logger.service.spec.ts

```diff
@@ -19,6 +23,14 @@ describe('ExpressLoggerService', () => {
   let logger: ExpressServerLogger;
   let loggerService: ExpressLoggerService;
 
+  beforeAll(() => {
+    TestBed.initTestEnvironment(
+      BrowserDynamicTestingModule,
+      platformBrowserDynamicTesting(),
+      {}
+    );
+  });
```

Comment:
[QUESTION]
Could you help me understand the reason for adding `BrowserDynamicTestingModule` and `platformBrowserDynamicTesting()` in the unit test setup for SSR?

I have probvlem to undestand why we include browser-related utils, while the code-under-test is meant only to run in Node (SSR). Also the tests deliberately run in Node (jest).

Reply:
Such a config is provided by `jest-preset-angular/setup-jest` [(source code)](https://github.com/thymikee/jest-preset-angular/blob/85896348ca208f069a4e34c9b36f1b4b8b50ac7e/setup-jest.js) which we're using in our app for Jest. My assumption is each Jest test that uses Angular's  API e.g. DI needs such a configuration, otherwise, tests throw errors.

Before my changes, `jest-preset-angular/setup-jest` was imported in the `setup-jest.ts` file and it was set up before all tests in the `ssr/setup` library. Because the `TestBed` config mentioned above creates a new platform, it caused the issues with tests for `ngExpressEngine`:
`A platform with a different configuration has been created. Please destroy it first`

As the platform was already there (due to the mentioned config). To prevent such an error, I removed `TestBed.initTestEnvironment` from the global config for the whole library and added it to the specific tests where it was needed.

We can investigate in the future if there's another way to configure tests for the `ssr/setup` library, but for now, I haven't found another option and I believe the most important is to have working tests.

Reply:
I wouldn't call it a dirty workaround (or it's a workaround but we have had it for a while) - I checked which part of `jest-preset-angular/setup-jest` is causing the issue (by commenting code of this package in node_modules) and `TestBed.initTestEnvironment`  and this is something we need (and we have it in other places).

Related issue: https://github.com/SAP/spartacus/issues/18205

Reply:
What about using `ServerTestingModule` and `platformServerTesting` instead in all spec.ts files in `setup/ssr` library? e.g.

```ts
import {
  platformServerTesting,
  ServerTestingModule,
} from '@angular/platform-server/testing';

/* ... */

  beforeAll(() => {
    TestBed.initTestEnvironment(
      ServerTestingModule,
      platformServerTesting(),
      {}
    );
  });
```

Reply:
Nice finding! Additionally, `platformServerTesting` doesn't cause platform-related errors in unit tests for `ng-express-engine` so config initialization can be kept in `setup-jest.ts` for the whole library. What a simplification 👌